src/Package/MSBuild.VSSetup/files.swr

```diff
@@ -10,10 +10,12 @@ vs.dependencies
   vs.dependency id=Microsoft.VisualStudio.PackageGroup.NuGet
 
 vs.relatedProcessFiles
+  vs.relatedProcessFile Path="[InstallDir]\MSBuild\Current\Bin\MSBuild.exe"
```

Comment:
As I recall `relatedProcessFile` means "use the RestartManager API to see if this file is in use and if so complain to the user rather than failing to overwrite it during setup/upgrade.

Reply:
Yup, it didn't work.

@genlu, may I ask for your expert advice here? Does setting the `ApplyNgenOptimization` prop enable OptProf end-to-end or is there some other place where we indicate for which assemblies optimization data should be created?

This is our OptProf training pipeline: [internal link](https://dev.azure.com/devdiv/DevDiv/_release?definitionId=1370&view=mine&_a=releases)

Reply:
`ApplyNgenOptimization` doesn't do it, we need to enable generating optimization data for the assembly first.

```
##[error].packages\microsoft.dotnet.arcade.sdk\6.0.0-beta.21379.2\tools\OptimizationData.targets(111,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) No optimization data found for assemblies: C:\a\1\s\artifacts\obj\MSBuild\Release\net472\MSBuild.exe
```

Let's get this in then and work on MSBuild.exe separately.

Reply:
Sorry I missed your question, notification from msbuild repo is being filtered out from my inbox. Please let me know if there's anything else I could help :)

Reply:
@genlu yes, the question is still valid. How does the OptProf infrastructure determine for which assemblies it should create optimization data? We noticed that it's not being done for MSBuild.exe and we would like to fix it. Thank you!

Reply:
This is specified in our optprof [runsetting file](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/VS?path=/src/Tests/config/runsettings/Official/OptProf/RoslynOptProf/RoslynOptProf.runsettingsproj&version=GBmain&line=28&lineEnd=29&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents), which points to the [config files](https://dev.azure.com/dnceng/internal/_build/results?buildId=1366625&view=artifacts&pathAsName=false&type=publishedArtifacts) we update with each of our insertion.